cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -220,10 +199,13 @@ func (s *service) preStart(c *runc.Container) (handleStarted func(*runc.Containe
 			for _, ee := range ees {
 				s.handleProcessExit(ee, c, p)
 			}
-			for _, eee := range initExits {
-				for _, cp := range initCps {
-					s.handleProcessExit(eee, cp.Container, cp.Process)
+			if !init {
+				s.lifecycleMu.Lock()
+				s.runningExecs[c]--
+				if ch, ok := s.execCountSubscribers[c]; ok {
+					ch <- s.runningExecs[c]
 				}
+				s.lifecycleMu.Unlock()
 			}
```

Comment:
```suggestion
```
Another copy of this code is present in `handleProcessExit`. We are going to have a bad time if we decrement `runningExecs` twice for each exec exit.

Reply:
You're right, good catch.